File: docs/guides/csharp/src/saveObjectsChunks.cs
namespace Algolia;

using System;
using System.Collections.Generic;
using System.Net.Http;
using System.Text.Json;
using Algolia.Search.Clients;
using Algolia.Search.Http;
using Algolia.Search.Models.Search;

class SaveObjectsChunks
{
  async Task Main(string[] args)
  {
    try
    {
      var client = new SearchClient(new SearchConfig("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY"));
      var jsonContent = await File.ReadAllTextAsync("actors.json");
      var records = JsonSerializer.Deserialize<List<Dictionary<string, object>>>(jsonContent);

      const int chunkSize = 10000;

      for (var beginIndex = 0; beginIndex < records?.Count; beginIndex += chunkSize)
      {
        var chunk = records.Slice(beginIndex, Math.Min(beginIndex + chunkSize, records.Count));
        await client.SaveObjectsAsync("<YOUR_INDEX_NAME>", chunk);
      }
    }
    catch (Exception e)
    {
      Console.WriteLine(e.Message);
    }
  }
}

File: docs/guides/go/src/saveObjectsChunks.go
package main

import (
	"context"
	"encoding/json"
	"os"

	"github.com/algolia/algoliasearch-client-go/v4/algolia/next/search"
)

func saveObjectsChunks() {
	client, err := search.NewClient("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")
	if err != nil {
		// The client can fail to initialize if you pass an invalid parameter.
		panic(err)
	}

	content, err := os.ReadFile("actors.json")
	if err != nil {
		panic(err)
	}

	var records []map[string]any

	err = json.Unmarshal(content, &records)
	if err != nil {
		panic(err)
	}

	chunkSize := 10000

	for beginIndex := 0; beginIndex < len(records); beginIndex += chunkSize {
		chunk := records[beginIndex:min(beginIndex+chunkSize, len(records))]

		_, err = client.SaveObjects(context.Background(), "<YOUR_INDEX_NAME>", chunk)
		if err != nil {
			panic(err)
		}
	}
}
